src/cpp/r/RSexp.cpp

```diff
@@ -172,24 +196,58 @@ SEXP findVar(const std::string& name, const std::string& ns)
    
    return findVar(name, env);
 }
-  
+
 SEXP findFunction(const std::string& name, const std::string& ns) 
 {
+   r::sexp::Protect protect;
```

Comment:
Wondering what the impetus for this change was? I'm thinking it might be to be able to unroll the PROMSXP or perhaps to take advantage of the global cache for R_GlobalEnv. This is clearly a change that poses some regression risk so I just want to make sure we absolutely need it.